Release 0.4.0 (#691)
publiccodenet/standard: criteria/findability.md

---
order: 14
---

# Make the codebase findable

<!-- SPDX-License-Identifier: CC0-1.0 -->
<!-- written in 2022 by The Foundation for Public Code <info@publiccode.net> -->

## Requirements

> * The codebase MUST be findable using a search engine by description of the problem it solves in natural language.

Reviewer: ```suggestion
* The codebase MUST be findable using a search engine by describing the problem it solves in natural language.
```

* The codebase MUST be findable using a search engine by codebase name.
* Maintainers SHOULD submit the codebase to relevant software catalogs.
* The codebase SHOULD have a website which describes the problem the codebase solves using the preferred jargon of different potential users of the codebase (including technologists, policy experts and managers).
* The codebase SHOULD have a unique and persistent identifier where the entry mentions the major contributors, repository location and website.
* The codebase SHOULD include a machine-readable metadata description, for example in a [publiccode.yml](https://github.com/publiccodeyml/publiccode.yml) file.
* A dedicated domain name for the codebase is OPTIONAL.

> * Regular presentations at conferences by the community is OPTIONAL.

Reviewer: ```suggestion
* Regular presentations at conferences by the community are OPTIONAL.
```

## Why this is important

* Proactiveness is needed; just publishing a codebase and hoping it's found doesn't work.
* A metadata description file increases discoverability.
* The more findable a codebase is, the more potential new collaborators will find it.
* By having well-written metadata containing a unique and persistent identifer, such as a Wikidata item or FSF software directory listing, and thus being part of the semantic web, it will be easier for other people to refer, cite, disambiguate and discover the codebase through third party tools.

## What this does not do

> * Ensure new collaborators or replicators

Reviewer: ```suggestion
* Ensure new collaborators or replicators.
```

## How to test

Check that the codebase appears in the first set of results on more than one major search engine when you search for:

> * plain problem description

Reviewer: ```suggestion
* plain English problem description
```

> * plain problem description

Reviewer: "plain" reads as if we mean it in opposition to 'unadorned' - I think we actually mean simple, not plain English?

```suggestion
* simple problem description
```

Reviewer 2: We have a criterion named "Use plain English" ....

It may not be simple, but it must be clear.
